### DISCUSSION TITLE
add shortcut for raw sql function

### DISCUSSION THREAD

**Discussion by IlyaSemenov** [romeerez/orchid-orm#98] [Ideas]
As a developer, I would like to reduce boilerplate:

```ts
const value = 'new name';
const updatedCount = await db.my_long_table_name
  .find(1)
  .updateRaw(db.my_long_table_name.sql`name = ${value}`);
```

can become:

```ts
const value = 'new name';
const updatedCount = await db.my_long_table_name
  .find(1)
  .updateRaw(db.$sql`name = ${value}`);
```

I am not a big fan of magic prefixed methods, but I don't see why the sql shortcut goes to the table object either (it is not coupled with a particular table, or is it?).

Alternatively, we can have it objection style with `sql` is exported at module level:

```ts
import { sql } from "pqb"

const value = 'new name';
const updatedCount = await db.my_long_table_name
  .find(1)
  .updateRaw(sql`name = ${value}`);
```

**Comment by romeerez** [comment]
It's coupled to the table only for the case when calling it in the form with a column type:

```ts
db.table.select({
  num: db.table.sql((t) => t.integer())`2 + 2`
})
```

So the result knows the type and has the parser if the column has it. Otherwise, yes, it could be `db.$sql`. And `updateRaw` could accept a template string.

**Reply by IlyaSemenov** [reply]
So essentially the reason `t` is coupled to `db.table` is that `t` can pull `columnTypes` from `BaseTable` of `db.table`?

**Reply by romeerez** [reply]
Yes, db instance on itself doesn't have column types, so they are going through the tables.
